docs/get-started/cli.md

```diff
@@ -4,4 +4,219 @@ title: On your local machine
+
+Your directory should now contain the following files:
+
+* `formulas.ts` - A place to define the Coda formulas, including those that power sync tables and column formats.
+* `manifest.ts` - Defines the metadata and contents of your Pack.
+* `node_modules` - The dependencies downloaded from NPM (standard for Node.js projects).
+* `package-lock.json` - The versions of the dependencies downloaded from NPM (standard for Node.js projects).
+* `package.json` - The project's dependencies from NPM (standard for Node.js projects).
+* `schemas.ts` - A place to define the schemas (structured data types) used by your Pack.
+* `types.ts` - A place to define TypeScript types for the data used by your Pack.
+
+## Add functionality to you Pack
+
+Now that you've got the basic structure of a Pack it's time to customize it and add some functionality. In this tutorial you'll be creating a simple "Hello World" Pack with a single formula.
+
+1. Edit `formulas.ts` to include the definition of a new "Hello" formula:
+
+    === "formulas.ts"
+        ```ts hl_lines="4-7 10-34"
+        --8<-- "hello_cli/formulas.ts"
+        ```
+
+1. Edit `manifest.ts` to set the formula namespace to `HelloWorld`:
+
+    === "manifest.ts"
+        ```ts hl_lines="8"
+        --8<-- "hello_cli/manifest.ts"
+        ```
+
+    !!! note
+        The value in `formulaNamespace` isn't normally visible to the user, as Coda prefers to show the Pack's icon instead. However when a formula is copied from the formula editor and pasted as plain text the namespace is used instead of the icon.
+
+## Test your Pack locally
+
+One of the advantages of developing locally is that you can test your Pack code without having to upload it to Coda's servers. Let's test the new `Hello` formula you just added:
+
+```shell
+npx coda execute manifest.ts Hello "world"  
+```
+
+If everything works correctly this should output `Hello world!`.
+
+## Upload your Pack
+
+So far everything you've built only exists on your local machine, and Coda has no knowledge of it. To see it working in a real doc you'll need to upload your Pack to Coda's servers.
+
+### Register an API token
+
+The `coda` CLI uses the Coda API under the hood to upload your code, and likewise needs an API token to operate. Registering an API token is a one-time setup step that must be done for each new Pack.
+
+1. Register an API key for Pack uploads:
+
+    ```shell
+    npx coda register --codaApiEndpoint=https://{{coda.domain}}
+    ```
+
+1. When prompted to create a new API token, type `y` and hit enter.
+
+    This will open your browser to the API token creation dialog.
+
+1. In the **Name** field enter "Hello World Pack" and then click **Generate API token**.
+
+1. Scroll down the page to the **Coda API tokens** section, find the token you just created, and click the **Copy token** link.
```

Comment:
Ugh, we should fix this long-standing bug so you don't have to scroll. Mind filing that at go/packs?

Reply:
Done: https://go/packsbug/196